pinot-spi/src/main/java/org/apache/pinot/spi/config/table/IndexingConfig.java

```diff
@@ -191,6 +192,14 @@ public void setStarTreeIndexConfigs(List<StarTreeIndexConfig> starTreeIndexConfi
     _starTreeIndexConfigs = starTreeIndexConfigs;
   }
 
+  public boolean isAllowStarTreeCreationDuringSegmentLoad() {
```

Comment:
why do we need this config. can we put this at the cluster level or something?

also, if we really need this, rename the variable to enableDynamicStarTreeCreation

Reply:
We need this config on the table level so that when user adds star-tree indexes for a table, they can choose to create star-trees on server side or Hadoop/Spark side for backfill (or minion side in the future) based on the server load.
Renamed the config